### ISSUE TITLE
Support: FOREIGN KEY with REFERENCES to own table

### ISSUE THREAD

**Issue by cyrille-arundo** [oguimbal/pg-mem#65]
The following schem works:
```sql
CREATE TABLE my_table (
    id text NOT NULL PRIMARY KEY,
    field1 text,
    field2 text NOT NULL,
    parent_id text,    
);
```

This one fails:
```sql
CREATE TABLE my_table (
    id text NOT NULL PRIMARY KEY,
    field1 text,
    field2 text NOT NULL,
    parent_id text,    
    CONSTRAINT fk_id FOREIGN KEY (parent_id) REFERENCES my_table (id)
);

relation "my_table" does not exist

🐜 This seems to be an execution error, which means that your request syntax seems okay,
but the resulting statement cannot be executed → Probably not a pg-mem error.

*️⃣ Failed SQL statement: CREATE TABLE my_table (
    id text NOT NULL PRIMARY KEY,
    field1 text,
    field2 text NOT NULL,
    parent_id text,    
    CONSTRAINT fk_id FOREIGN KEY (parent_id) REFERENCES my_table (id)
);
;
```

**Comment by oguimbal** [comment]
This one is failing with postgres 😀  The behaviour of pg-mem is actually the expected behaviour 🎉

![image](https://user-images.githubusercontent.com/8973947/108364193-11d50a00-71f6-11eb-9853-1dca7d1d0725.png)

**Comment by oguimbal** [comment]
Should be fixed in `pg-mem@1.7.6`

**Comment by cyrille-arundo** [comment]
Thanks for the fix @oguimbal. I verified it and it works as expected in our unit-tests.
